### PR TITLE
Nc voter addrs

### PR DESCRIPTION
**Pull request by JakeStevens**
This training data contains address from the NC voter files
that failed the parser. The dominant address pattern targeted
is addresses in which the Street Name is multiple words, for example:
113  BILLY S WHITE RD  DOBSON 27017
Nearly fifty of these NC addresses were used as training examples.

### CODE REVIEW

**Comment by jeancochrane** [comment]
Thanks for this PR @JakeStevens! I'll take a look at it later this afternoon.

File: training/nc_train.xml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  <AddressString><AddressNumber>820</AddressNumber> <StreetNamePreDirectional>North</StreetNamePreDirectional> <StreetNamePreType>Highway</StreetNamePreType> <StreetName>71</StreetName> <StreetName>Business</StreetName> <PlaceName>Lowell</PlaceName> <StateName>AR</StateName> <ZipCode>72745</ZipCode></AddressString>
+  <AddressString><AddressNumber>100</AddressNumber> <StreetName>GENERAL</StreetName> <StreetName>A</StreetName> <StreetName>S</StreetName> <StreetName>JOHNSON</StreetName> <StreetNamePostType>ST</StreetNamePostType> <PlaceName>STANLEY</PlaceName> <ZipCode>28164</ZipCode></AddressString>
+  <AddressString><AddressNumber>2500</AddressNumber> <StreetNamePreDirectional>S</StreetNamePreDirectional> <NotAddress>AND</NotAddress> <StreetNamePreDirectional>W</StreetNamePreDirectional> <StreetName>FARM</StreetName> <StreetNamePostType>RD</StreetNamePostType> <PlaceName>HICKORY</PlaceName> <ZipCode>28602</ZipCode></AddressString>
```
Comment (jeancochrane) [inline training/nc_train.xml]: I'm wondering if the `AND` here should be an `IntersectionSeparator` ([per the docs](http://usaddress.readthedocs.io/en/latest/#details))? Seems like a confusing address, and I can't confirm its existence.

File: training/nc_train.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  <AddressString><AddressNumber>110</AddressNumber> <StreetName>ALEXANDER</StreetName> <StreetNamePostType>ST</StreetNamePostType> <OccupancyIdentifier>#</OccupancyIdentifier> <OccupancyType>APT</OccupancyType> <OccupancyIdentifier>3</OccupancyIdentifier> <PlaceName>ELIZABETHTOWN</PlaceName> <ZipCode>28337</ZipCode></AddressString>
+  <AddressString><AddressNumber>48</AddressNumber> <StreetName>ROBERT</StreetName> <StreetName>N</StreetName> <StreetName>BOWEN</StreetName> <StreetNamePostType>DR</StreetNamePostType> <PlaceName>WHITEVILLE</PlaceName> <ZipCode>28472</ZipCode></AddressString>
+  <AddressString><AddressNumber>4026</AddressNumber> <StreetName>MARK</StreetName> <StreetName>N</StreetName> <StreetName>SMITH</StreetName> <StreetNamePostType>RD</StreetNamePostType> <PlaceName>DEEP</PlaceName> <PlaceName>RUN</PlaceName> <ZipCode>28525</ZipCode></AddressString>
+  <AddressString><AddressNumber>1031</AddressNumber> <StreetName>21ST</StreetName> <StreetNamePostType>ST</StreetNamePostType> <StreetNamePostDirectional>NW</StreetNamePostDirectional> <StreetNamePostType>LN</StreetNamePostType> <PlaceName>HICKORY</PlaceName> <ZipCode>28601</ZipCode></AddressString>
```
Comment (jeancochrane) [inline training/nc_train.xml]: Is this a real address (and not a data entry error)? I can't find it with the `LN` token on [Google](https://goo.gl/maps/1uyP1W74FB32) or[ OSM](http://www.openstreetmap.org/search?query=1031%2021st%20street%20northwest%20hickory%20nc#map=18/35.74604/-81.37408). If it's real we can include it, but if it's an error then the duplicate `StreetNamePostType` shouldn't be included.

File: training/nc_train.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  <AddressString><AddressNumber>337</AddressNumber> <StreetName>HOBSON</StreetName> <StreetNamePostType>DR</StreetNamePostType> <OccupancyIdentifier>#</OccupancyIdentifier> <OccupancyType>APT</OccupancyType> <OccupancyIdentifier>5</OccupancyIdentifier> <PlaceName>MOCKSVILLE</PlaceName> <ZipCode>27028</ZipCode></AddressString>
+  <AddressString><AddressNumber>100</AddressNumber> <StreetName>PARK</StreetName> <StreetName>CIRCLE</StreetName> <StreetName>EAST</StreetName> <StreetNamePostType>RD</StreetNamePostType> <PlaceName>ELIZABETH</PlaceName> <PlaceName>CITY</PlaceName> <ZipCode>27909</ZipCode></AddressString>
+  <AddressString><AddressNumber>322</AddressNumber> <StreetNamePreDirectional>E</StreetNamePreDirectional> <StreetName>SUMTER</StreetName> <StreetNamePostType>ST</StreetNamePostType> <OccupancyIdentifier>#</OccupancyIdentifier> <OccupancyType>APT</OccupancyType> <OccupancyIdentifier>1</OccupancyIdentifier> <PlaceName>SHELBY</PlaceName> <ZipCode>28150</ZipCode></AddressString>
```
Comment (jeancochrane) [inline training/nc_train.xml]: Here's a good example of the `# APT <#>` pattern that I'm concerned about.

Comment (fgregg) [inline training/nc_train.xml]: I'm sorry but I can't quite figure out what pattern you are talking about.

File: training/nc_train.xml (hunk 2 above)
Comment (fgregg) [inline training/nc_train.xml]: while we don't want to train on made up address strings, we don't care about including addresses that actually exist.

File: training/nc_train.xml (hunk 3 above)
Comment (jeancochrane) [inline training/nc_train.xml]: No problem! In this case, the full string is:

`322 E SUMTER ST # APT 1 SHELBY 28150`

I'm concerned about the sequence of tokens `# APT 1`, which seems like a data entry error (`APT #1` would be the standard way of writing this, which places the two `OccupancyIdentifiers`, `#` and `1`, in the proper order).

Comment (fgregg) [inline training/nc_train.xml]: we welcome date entry errors.

Comment (JakeStevens) [inline training/nc_train.xml]: I agree that this seems like a strange data entry system, but it is common in the North Carolina Secretary of State voter rolls, many addresses have them that way. I do not know if this is common across other SoSs.

File: training/nc_train.xml (hunk 1 above)
Comment (JakeStevens) [inline training/nc_train.xml]: As stated, I cannot confirm its existence, only that multiple voters have indicated that they live on this street. On Zillow's website [here](https://www.zillow.com/homedetails/2320-S-And-W-Farm-Rd-Hickory-NC-28602/82177703_zpid/) it seems like is not an intersection of two streets, however.

File: training/nc_train.xml (hunk 2 above)
Comment (jeancochrane) [inline training/nc_train.xml]: Great! Thanks for the speedy replies to my comments, @JakeStevens and @fgregg. Sounds like we're all good, so I'm going to go ahead and pull this in.

**Comment by JakeStevens** [comment]
Out of curiosity, how are the training XML files taken care of post merge? Is NC kept in its own file? Added to the original training? Integrated into the model installed when using pip?

**Comment by jeancochrane** [comment]
We usually keep it as a separate file for testing, and then once we can confirm that performance hasn't depreciated we append the new addresses to `training/labeled.xml`, which acts as the master training file.

Things have been pretty backed up here thanks to the S3 outage, so I'll have the two files merged tomorrow afternoon once we can push builds again!

**Pull request merged**